src/solvers/refinement/string_constraint_generator.cpp

```diff
@@ -0,0 +1,1840 @@
+    return string_insert_float(expr);
+  else if(starts_with(id,cprover_string_insert_char_array_func))
+    return string_insert_char_array(expr);
+  else if(starts_with(id,cprover_string_substring_func))
+    return string_substring(expr);
+  else if(starts_with(id,cprover_string_trim_func))
+    return string_trim(expr);
+  else if(starts_with(id,cprover_string_to_lower_case_func))
+    return string_to_lower_case(expr);
+  else if(starts_with(id,cprover_string_to_upper_case_func))
+    return string_to_upper_case(expr);
+  else if(starts_with(id,cprover_string_char_set_func))
+    return string_char_set(expr);
+  else if(starts_with(id,cprover_string_value_of_func))
+    return string_value_of(expr);
+  else if(starts_with(id,cprover_string_empty_string_func))
+    return empty_string(expr);
+  else if(starts_with(id,cprover_string_copy_func))
+    return string_copy(expr);
+  else if(starts_with(id,cprover_string_of_int_func))
+    return of_int(expr);
+  else if(starts_with(id,cprover_string_of_int_hex_func))
+    return of_int_hex(expr);
+  else if(starts_with(id,cprover_string_of_float_func))
+    return of_float(expr);
+  else if(starts_with(id,cprover_string_of_double_func))
+    return of_double(expr);
+  else if(starts_with(id,cprover_string_of_long_func))
+    return of_long(expr);
+  else if(starts_with(id,cprover_string_of_bool_func))
+    return of_bool(expr);
+  else if(starts_with(id,cprover_string_of_char_func))
+    return of_char(expr);
+  else if(starts_with(id,cprover_string_of_char_array_func))
+    return of_char_array(expr);
+  else if(starts_with(id,cprover_string_set_length_func))
+    return string_set_length(expr);
+  else if(starts_with(id,cprover_string_delete_func))
+    return string_delete(expr);
+  else if(starts_with(id,cprover_string_delete_char_at_func))
+    return string_delete_char_at(expr);
+  else if(starts_with(id,cprover_string_replace_func))
+    return string_replace(expr);
+  else if(starts_with(id,cprover_string_data_func))
+    return string_data(expr);
+  else
+    {
+      std::string msg("string_exprt::function_application: unknown symbol :");
+      msg+=id.c_str();
+      throw msg;
+    }
+}
+
+
+irep_idt extract_java_string(const symbol_exprt & s)
+{
+  std::string tmp(s.get(ID_identifier).c_str());
+  std::string value = tmp.substr(31);
+  return irep_idt(value);
+}
+
+string_exprt string_constraint_generatort::string_constant(irep_idt sval, int char_width, unsignedbv_typet char_type){
+
+  string_exprt res(char_type);
+  std::string str = sval.c_str();
+  // should only do this for java
+  std::wstring utf16 = utf8_to_utf16le(str);
+
+  for (std::size_t i = 0; i < utf16.size(); ++i) {
+    std::string idx_binary = integer2binary(i,STRING_SOLVER_INDEX_WIDTH);
+    constant_exprt idx(idx_binary, refined_string_typet::index_type());
+    std::string sval_binary=integer2binary((unsigned)utf16[i], char_width);
+    constant_exprt c(sval_binary,char_type);
```

Comment:
Done in https://github.com/diffblue/cbmc/pull/341/commits/4c5d3526d4a2644b2592693cf2f0e927be0fb0fb